syft/source/excluding_file_resolver.go

```diff
@@ -0,0 +1,71 @@
+package source
+
+import (
+	"io"
+	"os"
+)
+
+type excludeFn func(string, os.FileInfo) bool
+
+type excludingResolver struct {
+	delegate  FileResolver
+	excludeFn excludeFn
+}
+
+func NewExcludingResolver(delegate FileResolver, excludeFn excludeFn) FileResolver {
+	return &excludingResolver{
+		delegate,
+		excludeFn,
+	}
+}
+
+func (r *excludingResolver) FileContentsByLocation(location Location) (io.ReadCloser, error) {
+	return r.delegate.FileContentsByLocation(location)
+}
+
+func (r *excludingResolver) FileMetadataByLocation(location Location) (FileMetadata, error) {
+	return r.delegate.FileMetadataByLocation(location)
+}
+
+func (r *excludingResolver) HasPath(path string) bool {
+	return r.delegate.HasPath(path)
+}
```

Comment:
Just curious, should filter logic come into play with these methods, too? Like if the caller is getting file content, or asking if the path exists, and the path should be filtered, should we just say "no"?

Reply:
it isn't _exactly_ necessary based on how this is used, but yes, it probably should also apply here